### ISSUE TITLE
AttributeError: 'long' object has no attribute 'replace'

### ISSUE THREAD

**Issue by cowanh00** [pucherot/Pi.Alert#6]
When performing the test alert scan, I get the following error.

```
pi@raspberrypi:~ $ python ~/pialert/back/pialert.py 1

Pi.Alert 2.52 (2021-01-11)
---------------------------------------------------------
Scan Devices
    ScanCycle: 1
    Timestamp: 2021-01-12 15:46:00

Scanning...
    arp-scan Method...
    Pi-hole Method...
    DHCP Leases Method...

Procesising scan results...
    Devices Detected.......: 17
        arp-scan Method....: 17
        Pi-hole Method.....: +0
        New Devices........: 0

    Devices in this cycle..: 17
        Down Alerts........: 0
        New Down Alerts....: 0
        New Connections....: 0
        Disconnections.....: 0
        IP Changes.........: 0

Updating DB Info...
    Sessions Events (connect / discconnect) ...
    Creating new devices...
    Updating Devices Info...
Traceback (most recent call last):
  File "/home/pi/pialert/back/pialert.py", line 1313, in <module>
    sys.exit(main())       
  File "/home/pi/pialert/back/pialert.py", line 72, in main
    res = scan_network()
  File "/home/pi/pialert/back/pialert.py", line 379, in scan_network
    update_devices_data_from_scan ()
  File "/home/pi/pialert/back/pialert.py", line 827, in update_devices_data_from_scan
    vendor = query_MAC_vendor (device['dev_MAC'])
  File "/home/pi/pialert/back/pialert.py", line 296, in query_MAC_vendor
    mac = pMAC.replace (':','')
AttributeError: 'long' object has no attribute 'replace'
```

**Comment by pucherot** [comment]
What python version are you using?
~~~
python -V
~~~

**Comment by cowanh00** [comment]
```
pi@raspberrypi:~ $ python -V
Python 2.7.16
```

**Comment by pucherot** [comment]
Please, can you run:
~~~
sqlite3 ~/pialert/db/pialert.db "SELECT dev_MAC FROM DEVICES WHERE dev_Vendor = '(unknown)' OR dev_Vendor ='' OR dev_Vendor IS NULL "
~~~

**Comment by cowanh00** [comment]
```
pi@raspberrypi:~ $ sqlite3 ~/pialert/db/pialert.db "SELECT dev_MAC FROM DEVICES WHERE dev_Vendor = '(unknown)' OR dev_Vendor ='' OR dev_Vendor IS NULL "
92:97:92:23:2d:b1
b2:be:76:97:10:9b
be:25:bd:0a:9a:d6
b2:be:76:fa:0c:6a
b2:be:76:f4:de:10
2808461089
3409513159
0
```

**Comment by pucherot** [comment]
OK, seems that there are some incorrect MAC.
Can you run this?
~~~
sqlite3 ~/pialert/db/pialert.db "SELECT * FROM CurrentScan"
~~~

**Comment by cowanh00** [comment]
Here you go:
```
pi@raspberrypi:~ $ sqlite3 ~/pialert/db/pialert.db "SELECT * FROM CurrentScan"
15|44:ad:b1:8a:69:7f|192.168.1.1|Sagemcom Broadband SAS|arp-scan
15|dc:a6:32:12:38:ba|192.168.1.16|Raspberry Pi Trading Ltd|arp-scan
15|44:8a:5b:d4:47:8b|192.168.1.19|Micro-Star INT'L CO., LTD.|arp-scan
15|bc:60:a7:65:b7:21|192.168.1.69|Sony Interactive Entertainment Inc.|arp-scan
15|b8:27:eb:9d:3e:91|192.168.1.101|Raspberry Pi Foundation|arp-scan
15|b2:be:76:be:b2:ee|192.168.1.12|(Unknown)|arp-scan
15|a8:60:b6:08:7c:ad|192.168.1.111|Apple, Inc.|arp-scan
15|b2:be:76:d9:3e:86|192.168.1.13|(Unknown)|arp-scan
15|00:2b:67:a6:a6:79|192.168.1.109|LCFC(HeFei) Electronics Technology co., ltd|arp-scan
15|00:e0:4c:53:44:58|192.168.1.123|REALTEK SEMICONDUCTOR CORP.|arp-scan
15|f0:79:60:2b:87:e6|192.168.1.60|Apple, Inc.|arp-scan
15|40:b4:cd:bb:66:23|192.168.1.31|Amazon Technologies Inc.|arp-scan
15|44:c6:5d:98:b6:af|192.168.1.135|Apple, Inc.|arp-scan
15|a2:5e:77:7f:75:01|192.168.1.149|(Unknown)|arp-scan
15|60:09:c3:6c:1e:91|192.168.1.10|u-blox AG|arp-scan
15|86:0f:41:65:e0:ca|192.168.1.27|(Unknown)|arp-scan
1|92:97:92:23:2d:b1|192.168.1.59|(Unknown)|arp-scan
1|44:ad:b1:8a:69:7f|192.168.1.1|Sagemcom Broadband SAS|arp-scan
1|dc:a6:32:12:38:ba|192.168.1.16|Raspberry Pi Trading Ltd|arp-scan
1|44:8a:5b:d4:47:8b|192.168.1.19|Micro-Star INT'L CO., LTD.|arp-scan
1|40:b4:cd:bb:66:23|192.168.1.31|Amazon Technologies Inc.|arp-scan
1|bc:60:a7:65:b7:21|192.168.1.69|Sony Interactive Entertainment Inc.|arp-scan
1|b8:27:eb:9d:3e:91|192.168.1.101|Raspberry Pi Foundation|arp-scan
1|00:2b:67:a6:a6:79|192.168.1.109|LCFC(HeFei) Electronics Technology co., ltd|arp-scan
1|a8:60:b6:08:7c:ad|192.168.1.111|Apple, Inc.|arp-scan
1|b2:be:76:be:b2:ee|192.168.1.12|(Unknown)|arp-scan
1|92:36:c8:59:16:ab|192.168.1.25|(Unknown)|arp-scan
1|a2:5e:77:7f:75:01|192.168.1.149|(Unknown)|arp-scan
1|60:09:c3:6c:1e:91|192.168.1.10|u-blox AG|arp-scan
1|b2:be:76:d9:3e:86|192.168.1.13|(Unknown)|arp-scan
1|44:c6:5d:98:b6:af|192.168.1.135|Apple, Inc.|arp-scan
1|86:0f:41:65:e0:ca|192.168.1.27|(Unknown)|arp-scan
```

**Comment by pucherot** [comment]
Sorry for the delay.
I need two additional queries:
~~~~
sqlite3 ~/pialert/db/pialert.db "SELECT * FROM Pihole_Network"
sqlite3 ~/pialert/db/pialert.db "SELECT * FROM DHCP_Leases"
~~~~
 
Thanks.

**Comment by cowanh00** [comment]
Here you go:
```
pi@raspberrypi:~ $ sqlite3 ~/pialert/db/pialert.db "SELECT * FROM Pihole_Network"
dc:a6:32:12:38:ba|192.168.1.16|mediaserver.lan|Raspberry Pi Trading Ltd|1610521022
44:ad:b1:8a:69:7f|192.168.1.1|(unknown)|Sagemcom Broadband SAS|0
44:c6:5d:98:b6:af|192.168.1.135|Hughs-iPhone.lan|Apple, Inc.|1610468111
40:b4:cd:bb:66:23|fe80::42b4:cdff:febb:6623|(unknown)|Amazon Technologies Inc.|1610490785
b2:be:76:d9:3e:86|192.168.1.13|(unknown)||1610521161
a8:60:b6:08:7c:ad|192.168.1.111|Marias-Air.lan|Apple, Inc.|1610520272
00:2b:67:a6:a6:79|192.168.1.109|CYTKRAPCHyMxCv9.lan|LCFC(HeFei) Electronics Technology co., ltd|1610474700
86:0f:41:65:e0:ca|192.168.1.27|Galaxy-A40.lan||1610490740
44:8a:5b:d4:47:8b|192.168.1.19|(unknown)|Micro-Star INTL CO., LTD.|1610463574
a2:5e:77:7f:75:01|192.168.1.149|Hughs-Galaxy-Tab-A.lan||1610479949
b8:27:eb:b4:fe:36|192.168.1.11|(unknown)|Raspberry Pi Foundation|0
92:97:92:23:2D:B1|192.168.1.59|Hugh-s-Galaxy-S20.lan||1610521116
14:20:5E:38:B9:8D|192.168.1.50|MariaWalsiPhone.lan|Apple, Inc.|1610520079
00:09:DF:48:9B:1B|192.168.1.143|(unknown)|Vestel Elektronik San ve Tic. A.Ş.|1610520436
B2:BE:76:F4:DE:10|192.168.1.51|Hugh-s-Galaxy-S20.lan||1610520827
00:E0:4C:53:44:58|192.168.1.123|(unknown)|Realtek Semiconductor Corp.|1610491958
BC:60:A7:65:B7:21|192.168.1.69|(unknown)|Sony Interactive Entertainment Inc.|1610513566
B2:BE:76:97:10:9B|192.168.1.128|Hugh-s-Galaxy-S20.lan||1610455721
B8:EE:0E:09:01:43|192.168.1.17|TVbox.lan|Sagemcom Broadband SAS|1610520216
92:36:C8:59:16:AB|192.168.1.25|MariaWalsiPhone.lan||1610463306
80:1F:02:7E:7D:2E|192.168.1.110|cowanh00-desktop.lan|Edimax Technology Co. Ltd.|1610457477
f0:8a:76:09:75:46|192.168.1.86|Hughs-Galaxy-Tab-A.lan|Samsung Electronics Co.,Ltd|1610520436
60:09:c3:6c:1e:91|192.168.1.10|(unknown)|u-blox AG|1610519990
f0:79:60:2b:87:e6|192.168.1.60|(unknown)|Apple, Inc.|1610520249
ac:84:c6:0f:9d:84|192.168.1.53|HS100.lan|Tp-Link Technologies Co.,Ltd.|1610520007
pi@raspberrypi:~ $ sqlite3 ~/pialert/db/pialert.db "SELECT * FROM DHCP_Leases"
1610606394|ac:84:c6:0f:9d:84|192.168.1.53|HS100|01:ac:84:c6:0f:9d:84
1610606833|f0:8a:76:09:75:46|192.168.1.86|Hughs-Galaxy-Tab-A|01:f0:8a:76:09:75:46
1610606397|b8:ee:0e:09:01:43|192.168.1.17|*|*
1610526035|00:2b:67:a6:a6:79|192.168.1.109|CYTKRAPCHyMxCv9|01:00:2b:67:a6:a6:79
1610556997|92:36:c8:59:16:ab|192.168.1.25|*|01:92:36:c8:59:16:ab
1610606631|a8:60:b6:08:7c:ad|192.168.1.111|Marias-Air|01:a8:60:b6:08:7c:ad
1610606437|14:20:5e:38:b9:8d|192.168.1.50|MariaWalsiPhone|01:14:20:5e:38:b9:8d
1610606390|92:97:92:23:2d:b1|192.168.1.59|*|01:92:97:92:23:2d:b1
1610548371|44:8a:5b:d4:47:8b|192.168.1.19|cowanh00-desktop|01:44:8a:5b:d4:47:8b
1610524168|b2:be:76:97:10:9b|192.168.1.128|Hugh-s-Galaxy-S20|01:b2:be:76:97:10:9b
1610606399|b2:be:76:be:b2:ee|192.168.1.12|TL-WA855RE|01:b2:be:76:be:b2:ee
1610606384|60:09:c3:6c:1e:91|192.168.1.10|*|01:60:09:c3:6c:1e:91
1610606413|b2:be:76:d9:3e:86|192.168.1.13|*|*
1610581024|dc:a6:32:12:38:ba|192.168.1.16|mediaserver|01:dc:a6:32:12:38:ba
1610576772|86:0f:41:65:e0:ca|192.168.1.27|Galaxy-A40|01:86:0f:41:65:e0:ca
1610576603|bc:60:a7:65:b7:21|192.168.1.69|*|01:bc:60:a7:65:b7:21
1610544649|a2:5e:77:7f:75:01|192.168.1.149|*|01:a2:5e:77:7f:75:01
1610606356|00:09:df:48:9b:1b|192.168.1.143|*|01:00:09:df:48:9b:1b
1610600789|00:e0:4c:53:44:58|192.168.1.123|raspberrypi|01:00:e0:4c:53:44:58
1610551273|44:c6:5d:98:b6:af|192.168.1.135|Hughs-iPhone|01:44:c6:5d:98:b6:af
1610566738|b2:be:76:f4:de:10|192.168.1.51|*|01:b2:be:76:f4:de:10
1610606624|f0:79:60:2b:87:e6|192.168.1.60|*|01:f0:79:60:2b:87:e6
1610542816|40:b4:cd:bb:66:23|192.168.1.31|amazon-14c9e49bb|*
1610543865|80:1f:02:7e:7d:2e|192.168.1.110|*|01:80:1f:02:7e:7d:2e
1610606630|1|2a01:115f:765:8b00::152|*|00:01:00:01:25:6f:98:6a:f0:79:60:2b:87:e6
1610562642|2808461089|2a01:115f:765:8b00::171|*|00:03:00:01:bc:60:a7:65:b7:21
1610548374|3409513159|2a01:115f:765:8b00::12d|cowanh00-desktop|00:04:4a:2b:9c:21:5b:e0:9d:40:ba:54:78:ee:22:ab:ad:28
1610606627|0|2a01:115f:765:8b00::1d1|*|00:01:00:01:25:6f:98:6a:f0:79:60:2b:87:e6
```

**Comment by pucherot** [comment]
Solved in v2.54 (still not released)
Please update the file "pialert/back/pialert.py" and test.

**Comment by cowanh00** [comment]
This is working now. Thank you very much for creating this tool. It is very useful for me. :-)

```
pi@raspberrypi:~/pialert/back $ python ~/pialert/back/pialert.py 1

Pi.Alert 2.52 (2021-01-11)
---------------------------------------------------------
Scan Devices
    ScanCycle: 1
    Timestamp: 2021-01-13 12:06:00

Scanning...
    arp-scan Method...
    Pi-hole Method...
    DHCP Leases Method...

Procesising scan results...
    Devices Detected.......: 16
        arp-scan Method....: 16
        Pi-hole Method.....: +0
        New Devices........: 0

    Devices in this cycle..: 15
        Down Alerts........: 0
        New Down Alerts....: 0
        New Connections....: 0
        Disconnections.....: 1
        IP Changes.........: 0

Updating DB Info...
    Sessions Events (connect / discconnect) ...
    Creating new devices...
    Updating Devices Info...
    Voiding false (ghost) disconnections...
    Pairing session events (connection / disconnection) ...
    Creating sessions snapshot...
    Skipping repeated notifications...

Reporting...
    Formating report...
    Sending report by email...
    Notifications: 158

DONE!!!
```
